src/neural-network.js

```diff
@@ -35,12 +35,27 @@ export default class NeuralNetwork {
     };
   }
 
+  static get trainingValidations() {
+    return {
+      iterations: (val) => { return typeof val === 'number' && val > 0; },
+      errorThresh: (val) => { return typeof val === 'number' && val > 0 && val < 1; },
+      log: (val) => { return typeof val === 'function' || typeof val === 'boolean'; },
+      logPeriod: (val) => { return typeof val === 'number' && val > 0; },
```

Comment:
logPeriod only matters if you also defined `log` as well. I think these functions don't _have_ to be a one-to-one mapping for the training options and can instead explicitly check the training options they are testing. This way, they all would be of the form `(opts) => { ... }`.

Reply:
I am not sure I completely understand your point.  I could couple them so that we only check things that are dependent on each other and I could set them to check the value of the already set property.  I chose not to because I wanted to keep each property decoupled.  I think that will make it easier to extend (if you change the function of one property you don't have to worry about additional side effects) and I also didn't want to check every property each time I validate, just the properties that are changed.  Does that make sense?  I am open to changing but I don't yet understand the benefit.